### ISSUE TITLE
Concurrent mempool verification

### ISSUE THREAD

**Issue by sidhujag** [bitcoin/bitcoin#13140]
Have you guys ever considered a threadpool to verify mempool verifications to decrease latency of tx's propogated across the network? Obviously the biggest hurdle is the DDOS vector from invalid signatures but do you think there is a way around that through some sort of rate-limiting approach or something more clever possibly to detect and manage stray/invalid signature verification requests?

With the multi-core CPUs and highly optimized secp256k1 it may allow us to scale throughput drastically if it were possible, just a thought!

**Comment by sipa** [comment]
Transaction propagation isn't limited by validation speed, but intentionally slowed down by a few seconds to obscure transaction origins on the network.

It's certainly possible to optimize performance of transactions, but in practice this is very low priority. The global throughput is limited to a few thousand per 10 minutes anyway, which trivial to keep up with.

**Comment by sidhujag** [comment]
Hi Pieter,

Yes I know of the trickle thing I was wondering about that too. Yea it makes sense. In my case I am investigating to increase this throughput through multi-core. I have a proof of concept which works and is 2x-3x faster because it relays in parallel and bans if invalid sig. However obviously this won't work in a production environment because of the attacks so I thought I'd brainstorm with you guys!

**Comment by sipa** [comment]
But you can't increase throughput - it's limited by the block weight limit.

You can decrease propagation time, but that's undesirable as it will reveal more about transaction sourcing.

The only thing this will accomplish is decrease CPU usage a bit.

**Comment by MarcoFalke** [comment]
The feature request didn't seem to attract much attention in the past. Also, the issue seems not important enough right now to keep it sitting around idle in the list of open issues.

Closing due to lack of interest. Pull requests with improvements are always welcome.
